### PR TITLE
Fix stale svc clusterVIP fo multicluster

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**

Fix https://github.com/istio/istio/issues/39039

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [x] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: pilot/pkg/serviceregistry/aggregate/controller.go [hunk 1]
```diff
@@ -173,18 +174,24 @@ func (c *Controller) Services() []*model.Service {
...
 			for _, s := range svcs {
-				sp, ok := smap[s.Hostname]
+				previous, ok := smap[s.Hostname]
 				if !ok {
 					// First time we see a service. The result will have a single service per hostname
 					// The first cluster will be listed first, so the services in the primary cluster
 					// will be used for default settings. If a service appears in multiple clusters,
 					// the order is less clear.
-					smap[s.Hostname] = s
+					smap[s.Hostname] = index
```
Comment (ramaraochavali) [inline pilot/pkg/serviceregistry/aggregate/controller.go:183]: What is the use of this index. It is not clear to me?

File: pilot/pkg/serviceregistry/aggregate/controller.go [hunk 2]
```diff
@@ -173,18 +174,24 @@ func (c *Controller) Services() []*model.Service {
...
+					index++
 					services = append(services, s)
 				} else {
+					// If previous service has not been deep copied
+					if len(services[previous].ClusterVIPs.GetAddresses()) < 2 {
```
Comment (ramaraochavali) [inline pilot/pkg/serviceregistry/aggregate/controller.go:188]: can please explain in comments why we are checking < 2?

File: pilot/pkg/serviceregistry/aggregate/controller.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pilot/pkg/serviceregistry/aggregate/controller.go:183]: The function is appending a services slice,  the index is used to record the index of a svc, so when I need to merge this svc with a new svc, I can update the slice quicky

Comment (ramaraochavali) [reply pilot/pkg/serviceregistry/aggregate/controller.go:183]: i see

File: pilot/pkg/serviceregistry/aggregate/controller.go (hunk 2 above)
Comment (hzxuzhonghu) [reply pilot/pkg/serviceregistry/aggregate/controller.go:188]: OK, this is a optimization. To make deepcopy at most once for like N clusters. I will add comments

**Comment by ramaraochavali** [comment]
LGTM. Leaving it open for @howardjohn  to take a look and final approval

**Comment by hzxuzhonghu** [comment]
ping @stevenctl

**Review by stevenctl — APPROVED** [review]

**Pull request merged**